src/node_http2.cc

```diff
@@ -2379,6 +2416,42 @@ void Http2Stream::RefreshState(const FunctionCallbackInfo<Value>& args) {
   }
 }
 
+void Http2Session::AltSvc(int32_t id,
+                          const char* origin,
+                          size_t origin_len,
+                          const char* value,
+                          size_t value_len) {
+  Http2Scope h2scope(this);
+  nghttp2_submit_altsvc(session_, NGHTTP2_FLAG_NONE,
+                        id,
+                        reinterpret_cast<const uint8_t*>(origin), origin_len,
+                        reinterpret_cast<const uint8_t*>(value), value_len);
+}
+
+// Submits an AltSvc frame to the sent to the connected peer.
+void Http2Session::AltSvc(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+  Http2Session* session;
+  ASSIGN_OR_RETURN_UNWRAP(&session, args.Holder());
+
+  int32_t id = args[0]->Int32Value(env->context()).ToChecked();
+
+  // TODO(jasnell): This is not exactly right as these are supposed to
+  // be ASCII only.
```

Comment:
Is this TODO not currently solvable? We know for a fact that at least  the origin is ASCII-only.

Reply:
Yep, was already working on it :-)